### PR TITLE
[Form] Fix Autocomplete setData

### LINKED ISSUE #667: [Autocomplete] SetData in AutocompleteField
Hello everyone,
I have **unmapped** field with type `AutocompleteField`
inside `preSetData` event
i trying to `setData` to this is field:
```
$form->add('reference',ReferenceAutocompleteField::class,[
    'mapped' => false,
    'data' => $data
]);
```
but i found this is error:
`The form's view data is expected to be a "App\Entity\Reference", but it is a "string". You can avoid this error by setting the "data_class" option to null or by adding a view transformer that transforms "string" to an instance of "App\Entity\Reference".
`

### PR DESCRIPTION
**Pull request by Aziz403**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix symfony/ux#667
| License       | MIT

Bypass autocomplete field check the viewData

### CODE REVIEW

File: src/Symfony/Component/Form/Form.php [hunk 1]
```diff
@@ -304,7 +304,7 @@ public function setData(mixed $modelData): static
         if (!FormUtil::isEmpty($viewData)) {
             $dataClass = $this->config->getDataClass();
 
-            if (null !== $dataClass && !$viewData instanceof $dataClass) {
+            if (null !== $dataClass && !$viewData instanceof $dataClass && $this->name !=='autocomplete') {
```
Comment (yceruto) [inline src/Symfony/Component/Form/Form.php:307]: This change appears to be incorrect as the name of the form field is an arbitrary value.

**Review by yceruto — CHANGES_REQUESTED** [review]
> The form's view data is expected to be a "App\Entity\Reference", but it is a "string". You can avoid this error by setting the "data_class" option to null or by adding a view transformer that transforms "string" to an instance of "App\Entity\Reference". 

Resolving your issue can be achieved by following one of the two options outlined in that error message. Or you should pass an instance of `App\Entity\Reference` as `data` value instead of an string.

**Comment by Aziz403** [comment]
> Resolving your issue can be achieved by following one of the two options outlined in that error message. Or you should pass an instance of `App\Entity\Reference` as `data` value instead of an string.

its not about the exception, in this is situation shouldn't throw the error!
its a exception when using `ux-autocomplete`, we need to bypass check `$dataClass`.
check [this](https://github.com/https://github.com/symfony/ux/issues/667)

**Comment by derrabus** [comment]
Very well, but what you're doing here is you change a very generic form class and add a special case for a very specific control. I strongly believe that there has to be a better way to address your problem.

Oh, and we always need a test that covers your change.

**Comment by Aziz403** [comment]
You convinced me, and wow! I found a better solution that doesn't need to modify symfony/form.

**Pull request closed without merging**